src/algorithms/rsa.rs

```diff
@@ -245,6 +245,58 @@ pub fn recover_primes(n: &BigUint, e: &BigUint, d: &BigUint) -> Result<(BigUint,
     Ok((p, q))
 }
 
+/// Compute the modulus of a key from its primes.
+pub(crate) fn compute_modulus(primes: &[BigUint]) -> BigUint {
+    let mut n = BigUint::one();
+
+    for prime in primes {
+        n *= prime;
+    }
+
+    n
+}
+
+/// Compute the private exponent from its primes (p and q) and public exponent
+pub(crate) fn compute_private_exponent(primes: &[BigUint], exp: &BigUint) -> Result<BigUint> {
+    if primes.len() < 2 {
+        return Err(Error::InvalidPrime);
+    }
+
+    let mut totient = BigUint::one();
+
+    for prime in primes {
+        totient *= prime - BigUint::one();
+    }
+
+    if let Some(d) = exp.mod_inverse(totient) {
+        Ok(d.to_biguint().unwrap())
+    } else {
+        // `exp` evenly divides `totient`
+        Err(Error::InvalidPrime)
+    }
+}
```

Comment:
This somewhat duplicates https://github.com/RustCrypto/RSA/blob/c7b0eae/src/algorithms/generate.rs#L93-L113

Reply:
I tried something to reuse those bits. Let me know what you think.

Reply:
or make generate reuse those bits, I guess.